dts/bindings/dma/dmamux-controller.yaml

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2019, STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+# Common fields for DMAMUX controllers
+
+include: base.yaml
+
+child-bus: dmamux
+
+properties:
+    label:
+      required: true
+
+    channel-nb:
+      type: int
+      required: true
+      description: Number of DMAMUX output request channels supported by the controller
+
+    gen-nb:
+      type: int
+      required: true
+      description: Number of DMAMUX Request generator supported by the controller
+
+    trigger-nb:
+      type: int
+      required: true
+      description: Number of DMAMUX request trigger input supported by the controller
+
+    sync-nb:
+      type: int
+      required: true
+      description: Number of DMAMUX synchronization inputs supported by the controller
+
+    req-nb:
+      type: int
+      required: true
+      description: Number of DMAMUX Peripheral Request inputs supported by the controller
+
+    "#dmamux-cells":
+      type: int
+      required: false
+      const: 4
+
+# Parameter syntax of stm32 follows the dmamux client dts syntax
+# in the Linux kernel declared in
+# https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/diff/Bindings/dma/stm32-dmamux.txt
+
+dmamux-cells:
+  - channel
+  - channel-config
+  - gen
+  - gen-config
+
+# channel: DMAMUX channel this value cannot exceed the <dmamux-channels>
+# channel-config: configuration of the selected DMAMUX channel request (DMAMUX_CxCR)
+# gen: DMAMUX request generator input resource ; this value is >=0 and cannot exceed <dmamux-gen-nb>
```

Comment:
'this value is >=0 and cannot exceed <dmamux-gen-nb>'

Either > 0, either cannot exceed dmamux-gen-nb-1.
Reading below line, maybe "0" has a specific meaning? If yes, can you elaborate?

Reply:
gen: DMAMUX request generator input resource cannot exceed  `dmamux-gen-nb - 1`